lib/compiler/src/beam_core_to_ssa.erl

```diff
@@ -368,6 +385,20 @@ expr(#c_call{anno=A,module=M0,name=F0,args=Cargs}, Sub, St0) ->
                            args=[M0,F0,cerl:make_list(Cargs)]},
             expr(Call, Sub, St)
     end;
+expr(#c_primop{anno=A0,name=#c_literal{val=debug_line},
+               args=Cargs}, Sub, St0) ->
+    {Args,Ap,St1} = atomic_list(Cargs, Sub, St0),
+    #b_set{anno=A1} = I0 = primop(debug_line, A0, Args),
+    {_,Alias} = Sub,
+    A = A1#{alias => Alias},
+    I = I0#b_set{anno=A},
+    St2 = case St1 of
+              #kern{beam_debug_info=false} ->
+                  St1#kern{beam_debug_info=true};
+              #kern{} ->
+                  St1
+          end,
```

Comment:
The compiler and JIT ought to optimize this under the hood, is there any particular reason that optimization doesn't kick in here?

```suggestion
    St2 = St1#kern{beam_debug_info=true},
```

Reply:
I didn't think about that optimization. Yes, it should kick in.